### PR TITLE
feat: MI Assign home screen (M2-7405)

### PR DESCRIPTION
**Pull request by farmerpaul**
- [x] Tests for the changes have been added
- [ ] Related documentation has been added / updated
- [ ] OSS packages added to MindLogger [open source credit page](https://mindlogger.atlassian.net/jira/servicedesk/projects/MLA/knowledge/articles/340623543?spaceKey=MLA)

### 📝 Description

🔗 [Jira Ticket M2-7405](https://mindlogger.atlassian.net/browse/M2-7405)

Adds support for displaying assigned activities and flows in the Web App, including:

- Displays new `TargetSubjectLabel` chip component on multi-informant assignments
- Lists multiple instances of `ActivityCard` for the same activity/flow when user has multiple assignments for that activity/flow
  - _Currently, an activity can both be set to `autoAssign: true` as well as have manual assignments, and the web app respects this possibility. In [a future Admin App ticket](https://mindlogger.atlassian.net/browse/M2-7604), UI restrictions will be imposed to make auto-assign and manual assignments mutually exclusive._
- Tracks distinct in-progress state of each activity/flow for each assignment
- For those activities/flows that are scheduled for one-time completion or on a daily schedule: Uses new `targetSubjectId` returned by the `/answers/applet/{applet_id}/completions` endpoint to compare  assignment has already been completed that day

Also fixes a couple missing translations and incorrect usage of i18next plural forms.

> [!TIP]
> Although many files were touched in this PR, many of those constituted tying `targetSubjectId` to activity progress in [this commit](https://github.com/ChildMindInstitute/mindlogger-web-refactor/pull/530/commits/60f907138f10f802a25054f7c8aa037f5ad71995), and on the whole were somewhat repetitive. Strongly typing the `targetSubject` and `targetSubjectId` props to be required prop (cannot be `undefined`) made it easier to make that change in a comprehensive way. I attempted to group other related changes into their own commits and it may be easiest to review [commit-by-commit](https://github.com/ChildMindInstitute/mindlogger-web-refactor/pull/530/commits).

### 📸 Screenshots

https://github.com/user-attachments/assets/6f7d37f6-ce17-45db-8654-06ab5dd50381

### 🪤 Peer Testing

> [!NOTE] 
> This feature depends on the `enableActivityAssign` feature flag. This should already be enabled in dev/local environments. It would also be worth testing this with that feature flag **disabled** to confirm that existing behaviour is preserved with the MI Assign feature disabled, by adding `features.enableActivityAssign = false` prior to the `return` statement [here in `useFeatureFlags`](https://github.com/ChildMindInstitute/mindlogger-web-refactor/blob/47b863f1ed86f17b5120e4754cb5d07a4b2fde4b/src/shared/utils/hooks/useFeatureFlags.ts#L16-L17).

Please refer to [ticket's AC](https://mindlogger.atlassian.net/browse/M2-7405) for the main peer testing steps.

In addition, after completing a multi-informant assigned activity, confirm that the POST to the `answers` endpoint includes the appropriate `targetSubjectId`. Also confirm that for self-reports, that property is not included.

### CODE REVIEW

File: src/shared/constants/routes.ts [hunk 1]
```diff
@@ -112,14 +117,25 @@ const ROUTES = {
...
+      const params = new URLSearchParams();
+      params.append('appletId', appletId);
+      params.append('eventId', eventId);
+      params.append('activityId', activityId);
+      if (flowId) params.append('flowId', flowId);
+      if (publicAppletKey) params.append('publicAppletKey', publicAppletKey);
+      if (targetSubjectId) params.append('targetSubjectId', targetSubjectId);
+
+      return `${ROUTES.autoCompletion.path}?${params.toString()}`;
+    },
```
Comment (qiushihe) [inline src/shared/constants/routes.ts:138]: 🤩

**Review by qiushihe — COMMENTED** [review]
Seems to be mostly adding an extra parameter/attribute to all the places, plus some renaming/refactoring of variables and types.

But the code looks good nonetheless 👍 

@farmerpaul please let me know when you're done with the tests, and I'll give this another go.

**Review by mbanting — COMMENTED** [review]
Amazing work @farmerpaul! Have not reviewed the code but performed a round of peer testing which was mostly successful:

✅ _Auto-assigned activities & flows display_

✅ _Manually assigned activities & flows don't display when not assigned_

✅ _Manually assigned activities & flows display when assigned_

✅ _Manually assigned activities & flows inherit the existing global schedule_

✅ _Manually assigned activities & flows inherit the existing individual schedule_

⚠️ _Assigned activity (non self-report) sees a badge with question counter_
- Expected "13 Questions" as specified in Figma. Actual "13 questions" (lower case 'q'). Is Figma incorrect?

⚠️ _Assigned flow (non self-report) sees a badge with activity counter_
- Expected same as above, "2 Activities" (couldn't find example in Figma but inferring from activities). Actual "2 activities" (lower case 'a'). Is Figma incorrect?

✅ _Multiple assignments for the same activity / flow displayed for each subject_

✅ _Sorting available->scheduled, activity/flow order in admin panel, self report -> additional subjects_

⚠️ _Each assignment preserves its own unique in-progress state_
- Works for self-reports, but could not test with MI assigned activities. Starting activity results in BE returning 403 on call to `/subjects/{_id}` endpoint. Is this to be implemented in [M2-7475](https://mindlogger.atlassian.net/browse/M2-7475) to skip calling this endpoint in favor of getting subject info from assignment?

⚠️ _One time completion, single assignment activity disappears after being completed_
- Works for self-reports, but could not test with MI assigned activities (same issue as above)
 
⚠️ _One time completion, multiple assignment activity disappears after being completed (per assignment)_
- Works for self-reports, but could not test with MI assigned activities (same issue as above)

**Review by mbanting — APPROVED** [review]
Will leave to @qiushihe to complete their review of the code. [Peer tested](https://github.com/ChildMindInstitute/mindlogger-web-refactor/pull/530#pullrequestreview-2278732963) successfully 🎉 🎉 🎉

**Review by qiushihe — APPROVED** [review]
LGTM

**Pull request merged**